### PR TITLE
Llvm testing updates

### PR DESCRIPTION
**Pull request by daviditen**
Un-revert "Enable LLVM by default"

Update test scripts to use the system LLVM as much as possible

* Remove gcc/4.9 testing because LLVM requires at least gcc/5.1.

* We don't currently have LLVM on cygwin, so explicitly set CHPL_LLVM=none there.

* Use COMPILER_PATH= to tell clang where to find the standard libraries on
systems with PrgEnv-* modules.

Signed-off-by: David Iten <daviditen@users.noreply.github.com>

### CODE REVIEW

**Comment by daviditen** [comment]
The changes to third-party/llvm/Makefile and util/chplenv/chpl_llvm.py were from another PR that @mppf already reviewed, so they shouldn't need too close of a look. But happy to have your comments on them still.

**Comment by daviditen** [comment]
@ronawho, I'm still hoping you can take a look at this PR please.

**Comment by ronawho** [comment]
Whoops, missed the initial review request. I haven't been following the llvm work that well so I don't know how qualified I feel to review these types of PRs from a user perspective and for testing infrastructure changes we should be ramping up @Maxrimus.

File: util/cron/common-whitebox.bash [hunk 1]
```diff
@@ -128,15 +128,19 @@ fi
...
+# Make LLVM available. This has to happen after loading the PrgEnv or else
+# CCE's LLVM is added to PATHs etc. ahead of ours.
 if [ -z "${OFFICIAL_SYSTEM_LLVM}" ] ; then
   if [ -f /data/cf/chapel/setup_system_llvm.bash ] ; then
     source /data/cf/chapel/setup_system_llvm.bash
   elif [ -f /cray/css/users/chapelu/setup_system_llvm.bash ] ; then
     source /cray/css/users/chapelu/setup_system_llvm.bash
   fi
+  # Make clang aware of standard libraries
+  export COMPILER_PATH=/opt/gcc/10.3.0/snos
```
Comment (ronawho) [inline util/cron/common-whitebox.bash:140]: Why do we need this? These types of changes start to make me a little nervous. Nightly testing is meant to be testing things in a way that's close to how users would be using chapel, so I worry about changes like this.

I also have a slight worry about using a pre-built llvm instead of building from source since I suspect most users will have to build from source using `CHPL_LLVM=bundled`, but I don't really think we can afford to pay the build times in all our nightly llvm configuration so this is probably just what we have to deal with.

Comment (daviditen) [reply util/cron/common-whitebox.bash:140]: We need it because gcc libraries aren't in the normal Linux location on these systems, so clang needs a pointer to their base location. Without it, we get linking errors:
```bash
% source /cray/css/users/chapelu/setup_system_llvm.bash
% clang hello.c
/usr/bin/ld: cannot find crtbegin.o: No such file or directory
/usr/bin/ld: cannot find -lgcc
/usr/bin/ld: cannot find -lgcc_s
clang-11: error: linker command failed with exit code 1 (use -v to see invocation)
```

Comment (ronawho) [reply util/cron/common-whitebox.bash:140]: Is this something a user could hit, or is this something specific to our machines?

Comment (daviditen) [reply util/cron/common-whitebox.bash:140]: Actually I think it's because the LLVM build was done on a vanilla linux system, not on an XC/CS. If it were built on XC/CS it would have found them. So a user would only hit it if they're trying to share an LLVM build across multiple platforms.

Comment (ronawho) [reply util/cron/common-whitebox.bash:140]: Should we have a different system llvm that's built on XC/CS? (actually for XC we should build that as part of the XC module)

I'm ok with this as a temporary workaround, but it doesn't seem great long term. I also worry about hardcoding the path to `/opt/gcc/10.3.0`. Can we get this path from an env var?

File: util/cron/test-linux64-gcc49.bash [hunk 2]
```diff
@@ -1,19 +0,0 @@
-#!/usr/bin/env bash
```
Comment (Maxrimus) [inline util/cron/test-linux64-gcc49.bash:1]: So why was this file removed?

Comment (daviditen) [reply util/cron/test-linux64-gcc49.bash:1]: We stopped testing with gcc 4.9 because LLVM requires at least 5.1 to build, and our 4.9 environment doesn't work with llvm that was built with 5.x.

File: util/cron/common-whitebox.bash (hunk 1 above)
Comment (PHHargrove) [reply util/cron/common-whitebox.bash:140]: We've faced something similar.  So, I'll chime in here.

The gcc modules on an XC prepends`/opt/gcc/[VERSION]/snos/lib64` to LD_LIBRARY_PATH and similar for MAN_PATH and INFO_PATH, but doesn't provide an "isolated" env var with the setting.

We've dealt with this via the following mess:
```
hargrove@cori10:~> module load gcc
hargrove@cori10:~> echo $(dirname $(dirname $(g++ -print-file-name=libstdc++.so)))
/opt/gcc/8.3.0/snos/lib/gcc/x86_64-suse-linux/8.3.0/../../../..
```

Above is a Cray XC, but the result on a "normal" install still yields the proper path to use as, for instance, the `--gcc-toolchain` option to clang/clang++. 

Use of `gcc -print-libgcc-file-name` is another option, but requires stripping mode path components

Comment (PHHargrove) [reply util/cron/common-whitebox.bash:140]: I am a bit uncertain as to what the new expected behavior is now that this has been merged.  
Our CI for GASNet-EX is testing against chapel's `master` and since merge of this pull request fails with the following when running `make` to build the runtime (except when CC/CXX are clang/clang++):
```
gmake[1]: Entering directory `/home/phargrov/upcnightly/EX-ppc64el-smp-gcc-pshm/runtime/bld/dbg/chapel'
Making the compiler...
rm -rf /home/phargrov/upcnightly/EX-ppc64el-smp-gcc-pshm/runtime/bld/dbg/chapel/third-party/llvm/install/none
cd llvm && /usr/bin/gmake
Error: Please set the environment variable CHPL_LLVM to a supported value.
Supported values are:
  1) 'none' to build without LLVM support
  2) 'bundled' to build with the LLVM packaged in the third-party directory
  3) 'system' to use a pre-installed system-wide LLVM
See: https://chapel-lang.org/docs/latest/usingchapel/chplenv.html#chpl-llvm
gmake[4]: *** [none] Error 1
gmake[3]: *** [/home/phargrov/upcnightly/EX-ppc64el-smp-gcc-pshm/runtime/bld/dbg/chapel/third-party/llvm/install/none] Error 2
gmake[2]: *** [compiler] Error 2
gmake[1]: *** [comprt] Error 2
gmake[1]: Leaving directory `/home/phargrov/upcnightly/EX-ppc64el-smp-gcc-pshm/runtime/bld/dbg/chapel'
```

That all makes sense to me *except* that the proceeding `configure` seems to have successfully *defaulted* to `none`:
```

  Currently selected Chapel configuration:

CHPL_TARGET_PLATFORM: linux_ppc_le64
CHPL_TARGET_COMPILER: gnu
CHPL_TARGET_ARCH: ppc64le
CHPL_TARGET_CPU: unknown
CHPL_LOCALE_MODEL: flat
CHPL_COMM: gasnet *
  CHPL_COMM_SUBSTRATE: udp *
  CHPL_GASNET_SEGMENT: everything
CHPL_TASKS: qthreads
CHPL_LAUNCHER: amudprun
CHPL_TIMERS: generic
CHPL_UNWIND: none
CHPL_MEM: jemalloc
CHPL_ATOMICS: cstdlib
  CHPL_NETWORK_ATOMICS: none
CHPL_GMP: none
CHPL_HWLOC: bundled
CHPL_RE2: none
CHPL_LLVM: none
CHPL_AUX_FILESYS: none

  Selected installation options:

[...]
```

Is it really the intention that one _must_ now pick a `CHPL_LLVM` value explicitly?    
If so, what is the meaning/significance of the `configure` output (from `util/chplenv/printchplenv.py`) above?

Comment (daviditen) [reply util/cron/common-whitebox.bash:140]: > Is it really the intention that one must now pick a CHPL_LLVM value explicitly?

Yes, the intent was that you must explicitly pick `none` if that's really what you want. If we can detect that `bundled` is already built or that `system` will work we'll do that automatically, but you need to set it to `none` if that's your choice.

Comment (ronawho) [reply util/cron/common-whitebox.bash:140]: @daviditen can you provide some more context about our plans for llvm and what we're expecting from users now and for the next release. I have a hard time following what we're expecting from users, so it's probably not clear to external developers/collaborators.

Comment (daviditen) [reply util/cron/common-whitebox.bash:140]: The plan is to switch to using the LLVM backend by default instead of the backend that generates and compiles C code. This requires either installing a system llvm-11 with a command along the lines of `apt-get install llvm-11`, or building the bundled LLVM version by setting `CHPL_LLVM=bundled` and doing `make` from `$CHPL_HOME`. After the first bundled build is done the `CHPL_LLVM` variable is no longer required to use the bundled version.

If users prefer to continue using the C backend and avoid LLVM they will need to set `CHPL_LLVM=none`.

Comment (PHHargrove) [reply util/cron/common-whitebox.bash:140]: @daviditen 
Currently the majority of my CI builds just drop dead with the message I reported above.  
IIUC, the three ways for me to resolve this are to explicitly set `CHPL_LLVM=none` or `CHPL_LLVM=bundled`; or to install the prerequisites for use of `system` (which I could not find documented).  
I would really like to see at least some sketch of the planned user-facing documentation so that I can make the selection for my own CI platforms based on the same advice end users will see.  In particular I think I am potentially missing an entire step related to "bundled is _already_ built" (emphasis mine).

I will also repeat my confusion at having `configure` print `CHPL_LLVM: none` and the following `make` command tell me I must set it to one of a list of values which includes the same `none` which was reported by `configure`.

Comment (ronawho) [reply util/cron/common-whitebox.bash:140]: @PHHargrove I agree with your confusion. Could you open a github issue about documentation for llvm settings and for the configure confusion (linking to this thread)

Comment (PHHargrove) [reply util/cron/common-whitebox.bash:140]: Done: #17638
